rtrlib/lib/vector.h

```diff
@@ -14,45 +14,80 @@
 #include <stdlib.h>
 #include <string.h>
 
+/**
+ * @brief ASPA Object
+ * here the customer as number (CAS) authorizes a set of provider AS numbers
+ */
+struct aspas_record {
+	uint32_t customer_as_number;
+	size_t number_of_provider_as;
+	uint32_t *provider_as_numbers;
+};
+
 /**
  * @brief Struct which is similar in function to std::vector from c++ so a reallocating array
  */
-struct lrtr_vector {
+struct aspa_vector {
```

Comment:
doesn't the vector need a kind of locking mechanism or would we implement it on a higher level?

Reply:
I think thread safety is not the task of this vector.